### PR TITLE
GWRResultsLite class

### PR DESCRIPTION
**Pull request by Ziqi-Li**
Returning a lighter `GWRResultsLite` class in bandwidth searching, so it can use maths in `diagnostics.py`. 
I perfer using `_compute_betas_gwr`  for simplicity for now, since I found `spglm.iwls` is also doing global GLMs, and I am afraid of messing up with that.

### CODE REVIEW

File: gwr/gwr.py [hunk 1]
```diff
@@ -254,25 +254,47 @@ def fit(self, ini_params=None, tol=1.0e-5, max_iter=20, solve='iwls'):
...
+                        predy = np.dot(self.X[i],betas)[0]
+                        resid[i] = self.y[i] - predy
+                    elif isinstance(self.family, (Poisson, Binomial)):
+                        rslt = iwls(self.y, self.X, self.family, self.offset, None, ini_params, tol, max_iter, wi=wi)
+                        inv_xtx_xt = rslt[5]
+                        influ[i] = np.dot(self.X[i],inv_xtx_xt[:,i])*rslt[3][i][0]
+                        predy = rslt[1][i]
+                        resid[i] = self.y[i] - predy
+                return GWRResultsLite(self,resid,influ)
+
```
Comment (Ziqi-Li) [inline gwr/gwr.py:276]: below is same as what currently we have

**Comment by TaylorOshan** [comment]
Look ok to me. I don't quite follow your second point about ``_compute_betas_gwr`` versus ``spglm.iwls``...when you say messing with it, is that in reference to exploring sparse data structures?

Do you have any benchmark of the speedups of using the Lite results class versus not using it?

**Review by TaylorOshan — COMMENTED** [review]
Looks generally ok to me. What are on speed gains on this?

**Comment by Ziqi-Li** [comment]
I think `spglm.iwls` serves a more general purpose not only for gwr but also for other models. The only line needs for a gaussian gwr is [this](https://github.com/pysal/spglm/blob/b225342e941faf828d5cb071fe705c4ca57c30b1/spglm/iwls.py#L76), so I think its more straightforward to just call it.

**Comment by TaylorOshan** [comment]
Sure makes sense, I thought that was what you were already doing from your
previous optimization PR, so wasn't sure what the alternative would be. The
simpler, the better!

**Comment by Ziqi-Li** [comment]
Haven't done a benchmark, but after this PR, the expectation should be very similar to this #19, just made it more organized~

**Comment by TaylorOshan** [comment]
Ok, just trying to get a sense of where the speed-up comes from i.e.,
primarily from the avoidance of collecting as many model output or simply
from using _compute_betas_gwr instead of the full iwls routine.

**Comment by Ziqi-Li** [comment]
If total runtime is 185 seconds, it is like `return GWRResults()` takes ~35 secs, and `iwls` takes ~130 secs. After speedup, total is 27 seconds, it is like `_compute_betas_gwr` takes 24 secs, and return `GWRResultsLite()` takes nearly zero seconds. I would say major speedup from not calling `iwls` but returning a lighter output also matters.

**Comment by TaylorOshan** [comment]
Ok, sounds good!

**Pull request merged**